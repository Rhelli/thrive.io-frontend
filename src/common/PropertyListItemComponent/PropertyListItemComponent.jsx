import React from 'react';
import PropTypes from 'prop-types';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faMapMarkerAlt, faBed, faUsers, faBriefcase, faCog, faTrashAlt,
} from '@fortawesome/free-solid-svg-icons';
import { v4 as uuidv4 } from 'uuid';
import {
  occupationShorthand, noPropertiesMessageGen,
} from '../../utils/managePropertiesUtils';
import styles from './PropertyListItemComponent.module.scss';

const PropertyListItemComponent = ({
  properties, propertyClickThrough, handlePropertyOptionButton, listItemType,
}) => (
  <div className={styles.propertyListItemContainer}>
    {
      properties.length ? (
        properties.map(property => (
          <div key={uuidv4()}>
            <div
              className={styles.propertyContainer}
              onClick={() => propertyClickThrough(property)}
              role="button"
              onKeyUp={() => propertyClickThrough(property)}
              tabIndex="-1"
            >
              <div className={styles.propertyImage}>
                {
                  listItemType === 'manage' ? (
                    <button
                      type="button"
                      className={styles.propertySettingsButton}
                      onClick={property => handlePropertyOptionButton(property)}
                    >
                      <FontAwesomeIcon icon={faCog} />
                    </button>
                  ) : (
                    null
                  )
                }
                <h4>IMAGE</h4>
              </div>
              <div className={styles.propertyInfoContainer}>
                <div className={styles.propertyTitle}>
                  <h3>{property.title}</h3>
                  <span className={styles.propertyPrice}>
                    Â£
                    {property.price}
                  </span>
                </div>
                <div className={styles.propertyLocation}>
                  <span className={styles.propertyLocationIcon}>
                    <FontAwesomeIcon icon={faMapMarkerAlt} />
                  </span>
                  <span className={styles.propertyAddress}>
                    <p>
                      {property.address}
                      ,
                      {' '}
                      {property.town}
                    </p>
                  </span>
                </div>
                <div className={styles.propertyDetails}>
                  <span className={styles.propertyFreeRooms}>
                    <FontAwesomeIcon icon={faBed} />
                    <p>
                      {property.roomCount}
                      {' '}
                      Rooms
                    </p>
                  </span>
                  <span className={styles.propertyFlatmates}>
                    <FontAwesomeIcon icon={faUsers} />
                    <p>
                      {property.occupantCount}
                      {' '}
                      Flatmates
                    </p>
                  </span>
                  <span>
                    <FontAwesomeIcon icon={faBriefcase} />
                    <p>{occupationShorthand(property.occupations)}</p>
                  </span>
                </div>
              </div>
            </div>
            <div className={styles.shortlistDeleteButtonContainer}>
              {
                listItemType === 'Shortlist' ? (
                  <button type="button" onClick={() => handlePropertyOptionButton(property)}>
                    <p>Remove</p>
                    <FontAwesomeIcon icon={faTrashAlt} />
                  </button>
                ) : (
                  null
                )
              }
            </div>
          </div>
        ))
      ) : (
        listItemType === 'Manage' ? (
          <div className={styles.noPropertiesMessages}>
            <h2>There isn&apos;t anything here!</h2>
            <h4>{noPropertiesMessageGen()}</h4>
          </div>
        ) : (
          <div className={styles.noPropertiesMessages}>
            <h2>There isn&apos;t anything here!</h2>
            <h4>Have a browse of the available properties.</h4>
          </div>
        )
      )
    }
  </div>
);

PropertyListItemComponent.propTypes = {
  properties: PropTypes.arrayOf(
    PropTypes.shape({
      address: PropTypes.string.isRequired,
      bills: PropTypes.string.isRequired,
      blurb: PropTypes.string.isRequired,
      deposit: PropTypes.number.isRequired,
      disabledAccess: PropTypes.string,
      furnished: PropTypes.string.isRequired,
      genders: PropTypes.arrayOf(PropTypes.string),
      id: PropTypes.number,
      internet: PropTypes.string,
      maxAge: PropTypes.number,
      minAge: PropTypes.number,
      occupantCount: PropTypes.number.isRequired,
      occupations: PropTypes.arrayOf(PropTypes.string),
      outsideArea: PropTypes.arrayOf(PropTypes.string),
      parking: PropTypes.string.isRequired,
      pets: PropTypes.arrayOf(PropTypes.string),
      postcode: PropTypes.string.isRequired,
      price: PropTypes.number.isRequired,
      roomCount: PropTypes.number.isRequired,
      smoking: PropTypes.string,
      title: PropTypes.string.isRequired,
      town: PropTypes.string.isRequired,
    }).isRequired,
  ).isRequired,
  propertyClickThrough: PropTypes.func.isRequired,
  handlePropertyOptionButton: PropTypes.func.isRequired,
  listItemType: PropTypes.string.isRequired,
};

export default PropertyListItemComponent;
