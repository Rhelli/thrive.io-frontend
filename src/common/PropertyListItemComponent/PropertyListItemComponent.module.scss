.propertyListItemContainer {
  display: flex;
  flex-flow: column nowrap;
  width: 100%;
  height: auto;
  padding: 20px 20px 80px 20px;
  border: 1px solid green;
  z-index: 1;
}

.propertyContainer {
  display: flex;
  width: 100%;
  flex-flow: column nowrap;
  border: 1px solid red;
  z-index: 3;

  .propertyImage {
    display: flex;
    position: relative;
    justify-content: center;
    align-items: center;
    border: 1px solid blue;
    height: 160px;

    .propertySettingsButton,
    .propertyDeleteButton {
      display: flex;
      justify-content: center;
      align-items: center;
      position: absolute;
      top: 0;
      right: 0;
      padding: 8px 9px;
      outline: none;
      border: none;
      background-color: rgba(0, 100 , 230, 0.5); 
      border-radius: 50%;
      z-index: 10;

      svg {
        font-size: 18px;
      }
    }

    .propertyDeleteButton {
      background-color: rgba(230, 0, 0, 0.5);
    }
  }

  .propertyInfoContainer {
    .propertyTitle {
      display: flex;
      flex-flow: row nowrap;
      width: 100%;

      h3 {
        width: 85%;
        font-size: 18px;
        line-height: 1.2;
        border: 1px solid red;
        padding: 10px 0;
      }

      span {
        display: flex;
        padding: 10px 0;
        font-size: 18px;
      }
    }

    .propertyLocation {
      display: flex;
      flex-flow: row nowrap;
      align-items: center;
      padding: 10px 0;
      border: 1px solid green;

      .propertyLocationIcon {
        svg {
          font-size: 18px;
        }
        margin-right: 5px;
      }

      .propertyAddress {
        font-size: 16px;
      }

      .propertyPrice {
        margin-left: auto;
      }
    }

    .propertyDetails {
      border: 1px solid red;
      display: flex;
      flex-flow: row nowrap;
      align-items: center;
      justify-content: flex-start;
      padding: 10px 0;

      span {
        display: flex;
        flex-flow: row nowrap;
        align-items: center;
        margin-right: 10px;

        svg {
          position: relative;
          bottom: 3px;
          font-size: 18px;
          margin-right: 5px;
        }
      }
    }
  }
}

.propertyListContainer {
  .noPropertiesMessages {
    display: flex;
    width: 100%;
    flex-flow: column nowrap;
    align-items: center;
    justify-content: center;
    border: 1px solid red;
    padding: 40px 20px;
    
    h2,
    h4 {
      width: 100%;
      text-align: center;
    }

    h2 {
      font-size: 26px;
      line-height: 1.2;
      margin-bottom: 10px;
    }

    h4 {
      font-size: 18px;
      line-height: 1.2;
      font-weight: 200;
      font-style: italic;
    }
  
  }
}

.shortlistDeleteButtonContainer {
  display: flex;
  justify-content: center;
  align-items: center;
  width: 100%;
  padding: 5px 0 15px 0;

  button {
    display: flex;
    flex-flow: row nowrap;
    align-items: center;
    justify-content: center;
    height: 40px;
    width: 70%;
    border: none;
    outline: none;
    border-radius: 4px 4px 25px 25px;
    background-color: darkred;
    margin-bottom: 20px;

    p {
      margin-right: 8px;
      font-size: 16px;
    }

    svg {
      position: relative;
      font-size: 18px;
      bottom: 2px
    }
  }
}