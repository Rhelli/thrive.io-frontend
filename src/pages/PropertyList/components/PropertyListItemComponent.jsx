import React from 'react';
import PropTypes from 'prop-types';

const PropertyListItemComponent = props => {
  const {
    title, town, price, bills, occupantCount, roomCount,
  } = props;

  return (
    <div>
      <h3>{title}</h3>
      <h5>{town}</h5>
      <p>{price}</p>
      <p>{bills}</p>
      <p>{occupantCount}</p>
      <p>{roomCount}</p>
    </div>
  );
};

PropertyListItemComponent.propTypes = {
  title: PropTypes.string.isRequired,
  town: PropTypes.string.isRequired,
  price: PropTypes.number.isRequired,
  bills: PropTypes.bool.isRequired,
  occupantCount: PropTypes.number.isRequired,
  roomCount: PropTypes.number.isRequired,
};

export default PropertyListItemComponent;
