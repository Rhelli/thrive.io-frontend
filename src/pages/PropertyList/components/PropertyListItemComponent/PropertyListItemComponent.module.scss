.propertyListItemContainer {
  display: flex;
  flex-flow: column nowrap;
  justify-content: center;
  align-items: center;
  background-color: #ece2fd;
  margin-bottom: 12px;
  border-top-right-radius: 6px;
  border-top-left-radius: 6px;
  border-bottom: 6px solid #efa9fa;
  transition: 0.2s;

  &:hover,
  &:focus {
    opacity: 0.7;
    transition: 0.15s;
  }

  .propertyListItemImage {
    display: flex;
    width: 100%;
    height: 150px;
    justify-content: center;
    align-items: center;
  }

  .propertyListItemInfoContainer {
    display: flex;
    flex-flow: column nowrap;
    width: 100%;

    .propertyListItemTopInfo {
      display: flex;
      flex-flow: row nowrap;
      width: 100%;
      background-color: #c6a9fa;
      border-bottom: 1px solid #9f6ff7;

      span {
        display: flex;
        align-items: center;
        height: auto;
        padding: 10px;
        background-color: #c6a9fa;
        border-top: 1px solid #b38cf8;
      }

      span:first-child {
        font-size: 14px;
        width: 80%;
      }

      span:last-child {
        width: 20%;
        font-size: 16px;
        font-weight: 700;
        text-align: right;
        background-color: #b38cf8;
        border-left: 1px solid #9f6ff7;
        border-bottom-left-radius: 10px;
      }
    }

    .propertyListItemMiddleInfo {
      display: flex;
      flex-flow: row nowrap;
      width: 100%;

      span {
        width: 50%;
        padding: 10px;
      }

      span:last-child {
        text-align: right;
      }
    }

    .propertyListItemBottomInfo {
      width: 100%;
      padding: 10px;
    }
  }
}
