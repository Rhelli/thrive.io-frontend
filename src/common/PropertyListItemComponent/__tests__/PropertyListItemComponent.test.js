import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import PropertyListItemComponent from '../PropertyListItemComponent';

Enzyme.configure({ adapter: new Adapter() });

const properties = [
  {
    id: 1,
    title: 'Dolorem fuga provident.',
    blurb: 'Necessitatibus et ex et maiores provident dolorum est eius enim dignissimos sit magnam.',
    address: '152 Chanelle Villages',
    town: 'Brixton',
    postcode: 'SW18 9NB',
    price: 1050,
    deposit: 802,
    bills: 'Included',
    furnished: 'Furnished',
    parking: 'No Parking',
    outsideArea: [],
    disabledAccess: 'No Disabled Access',
    internet: 'Internet Included',
    occupantCount: 7,
    roomCount: 8,
    minAge: 18,
    maxAge: 51,
    smoking: 'Any',
    pets: ['None'],
    genders: ['Female', 'Transgender', 'Male', 'Female', 'Male', 'Transgender', 'Transgender'],
    occupations: ['Student', 'Professional', 'Student', 'Professional', 'Professional', 'Student', 'Student'],
  },
  {
    id: 2,
    title: 'Non in beatae unde nostrum hic.',
    blurb: 'Et unde dignissimos qui eveniet ipsa blanditiis quasi sunt dicta alias est ut eaque iusto excepturi ea voluptatem tempora officiis voluptatem sed amet.',
    address: '999 Graham Crossing',
    town: 'Bayswater',
    postcode: 'SW6 1PW',
    price: 1475,
    deposit: 1042,
    bills: 'Not Included',
    furnished: 'Furnished',
    parking: 'Parking',
    outsideArea: [],
    disabledAccess: 'No Disabled Access',
    internet: 'No Internet Included',
    occupantCount: 4,
    roomCount: 5,
    minAuge: 29,
    maxAuge: 59,
    smoking: 'Any',
    pets: ['Rodents'],
    genders: ['Male', 'Female', 'Male', 'Transgender'],
    occupations: ['Professional', 'Professional', 'Professional', 'Professional'],
  },
];
const noProperties = [];
const propertyClickThrough = jest.fn(() => 'Now on property page.');
const handlePropertyOptionButton = jest.fn(() => 'Property option button clicked');
const shortlistListItemType = 'Shortlist';
const manageListItemType = 'Manage';

const shortlistPropertyListItemComponent = (
  <PropertyListItemComponent
    properties={properties}
    propertyClickThrough={propertyClickThrough}
    handlePropertyOptionButton={handlePropertyOptionButton}
    listItemType={shortlistListItemType}
  />
);

const emptyShortlistPropertyListItemComponent = (
  <PropertyListItemComponent
    properties={noProperties}
    propertyClickThrough={propertyClickThrough}
    handlePropertyOptionButton={handlePropertyOptionButton}
    listItemType={shortlistListItemType}
  />
);

const managePropertyListItemComponent = (
  <PropertyListItemComponent
    properties={properties}
    propertyClickThrough={propertyClickThrough}
    handlePropertyOptionButton={handlePropertyOptionButton}
    listItemType={manageListItemType}
  />
);

describe('The appearance and functionality of the PropertyListItemComponent when it is in shortlist mode', () => {
  const container = mount(shortlistPropertyListItemComponent);
  const propertyList = container.find('.propertyList');
  const firstProperty = propertyList.at(0).childAt(0);

  it('Matches the previous snapshot of the shortlist PropertyListItemComponent', () => {
    const tree = renderer.create(shortlistPropertyListItemComponent).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Has a master container with the appropriate attributes', () => {
    const container = mount(shortlistPropertyListItemComponent);
    expect(container.childAt(0).name()).toBe('div');
    expect(container.childAt(0).hasClass('propertyListItemContainer')).toBe(true);
  });

  it('Iterates over and displays the correct number of shortlisted properties as passed in via the properties prop', () => {
    const container = mount(shortlistPropertyListItemComponent);
    expect(container.find('.propertyList').length).toBe(2);
  });

  it('Displays nothing but a friendly message if there are no properties in the shortlist', () => {
    const emptyContainer = mount(emptyShortlistPropertyListItemComponent);
    expect(emptyContainer.find('.propertyList').length).toBe(0);
    const errorMessages = emptyContainer.find('.noPropertiesMessages');
    expect(errorMessages.childAt(0).text()).toBe("There isn't anything here!");
  });

  it('Displays the property image element correctly', () => {
    expect(firstProperty.childAt(0).name()).toBe('div');
    expect(firstProperty.childAt(0).hasClass('propertyImage')).toBe(true);
    expect(firstProperty.childAt(0).childAt(0).text()).toBe('IMAGE');
  });

  it('Displays the property title and price correctly', () => {
    expect(firstProperty.childAt(1).hasClass('propertyInfoContainer')).toBe(true);
    expect(firstProperty.childAt(1).childAt(0).children().length).toBe(2);
    expect(firstProperty.childAt(1).childAt(0).childAt(0).text()).toBe('Dolorem fuga provident.');
    expect(firstProperty.childAt(1).childAt(0).childAt(1).text()).toBe('Â£1050');
  });

  it('Displays the property address and town correctly', () => {
    expect(firstProperty.childAt(1).childAt(1).hasClass('propertyLocation')).toBe(true);
    expect(firstProperty.childAt(1).childAt(1).childAt(0).hasClass('propertyLocationIcon'));
    expect(firstProperty.childAt(1).childAt(1).childAt(0).childAt(0).name()).toBe('FontAwesomeIcon');
    expect(firstProperty.childAt(1).childAt(1).childAt(1).hasClass('propertyAddress')).toBe(true);
    expect(firstProperty.childAt(1).childAt(1).childAt(1).childAt(0).text()).toBe('152 Chanelle Villages, Brixton');
  });

  it('Displays the property room and flatmate count correctly', () => {
    expect(firstProperty.childAt(1).childAt(2).hasClass('propertyDetails')).toBe(true);
    expect(firstProperty.childAt(1).childAt(2).childAt(0).childAt(0).name()).toBe('FontAwesomeIcon');
    expect(firstProperty.childAt(1).childAt(2).childAt(0).childAt(1).text()).toBe('8 Rooms');
    expect(firstProperty.childAt(1).childAt(2).childAt(1).childAt(0).name()).toBe('FontAwesomeIcon');
    expect(firstProperty.childAt(1).childAt(2).childAt(1).childAt(1).text()).toBe('7 Flatmates');
  });

  it('Formats and displays the occupations of the flatmates correctly', () => {
    expect(firstProperty.childAt(1).childAt(2).childAt(2).childAt(0).name()).toBe('FontAwesomeIcon');
    expect(firstProperty.childAt(1).childAt(2).childAt(2).childAt(1).text()).toBe('Student');
  });

  it('Displays a functioning shortlist delete button', () => {
    const deleteButton = container.find('.shortlistDeleteButtonContainer');
    expect(deleteButton.at(0).childAt(0).name()).toBe('button');
    deleteButton.at(0).childAt(0).simulate('click');
    expect(handlePropertyOptionButton).toHaveBeenCalledTimes(1);
  });
});

describe('The appearance and functionality of the PropertyListItem component when in manage mode', () => {
  const container = mount(managePropertyListItemComponent);
  const propertyList = container.find('.propertyList');
  const firstProperty = propertyList.at(0).childAt(0);

  it('Displays a functioning settings button in the property image', () => {
    expect(firstProperty.childAt(0).childAt(0).name()).toBe('button');
    expect(firstProperty.childAt(0).childAt(0).hasClass('propertySettingsButton')).toBe(true);
    firstProperty.childAt(0).childAt(0).simulate('click');
    expect(handlePropertyOptionButton).toHaveBeenCalledTimes(1);
  });
});

