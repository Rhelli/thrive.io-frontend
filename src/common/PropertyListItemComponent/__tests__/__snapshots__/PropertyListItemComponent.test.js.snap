// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The appearance and functionality of the PropertyListItemComponent when it is in shortlist mode Matches the previous snapshot of the shortlist PropertyListItemComponent 1`] = `
<div
  className="propertyListItemContainer"
>
  <div
    className="propertyList"
  >
    <div
      className="propertyContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="-1"
    >
      <div
        className="propertyImage"
      >
        <h4>
          IMAGE
        </h4>
      </div>
      <div
        className="propertyInfoContainer"
      >
        <div
          className="propertyTitle"
        >
          <h3>
            Dolorem fuga provident.
          </h3>
          <span
            className="propertyPrice"
          >
            £
            1050
          </span>
        </div>
        <div
          className="propertyLocation"
        >
          <span
            className="propertyLocationIcon"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-map-marker-alt fa-w-12 "
              data-icon="map-marker-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
          <span
            className="propertyAddress"
          >
            <p>
              152 Chanelle Villages
              ,
               
              Brixton
            </p>
          </span>
        </div>
        <div
          className="propertyDetails"
        >
          <span
            className="propertyFreeRooms"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-bed fa-w-20 "
              data-icon="bed"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M176 256c44.11 0 80-35.89 80-80s-35.89-80-80-80-80 35.89-80 80 35.89 80 80 80zm352-128H304c-8.84 0-16 7.16-16 16v144H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v352c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16v-48h512v48c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V240c0-61.86-50.14-112-112-112z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <p>
              8
               
              Rooms
            </p>
          </span>
          <span
            className="propertyFlatmates"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-users fa-w-20 "
              data-icon="users"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <p>
              7
               
              Flatmates
            </p>
          </span>
          <span>
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-briefcase fa-w-16 "
              data-icon="briefcase"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M320 336c0 8.84-7.16 16-16 16h-96c-8.84 0-16-7.16-16-16v-48H0v144c0 25.6 22.4 48 48 48h416c25.6 0 48-22.4 48-48V288H320v48zm144-208h-80V80c0-25.6-22.4-48-48-48H176c-25.6 0-48 22.4-48 48v48H48c-25.6 0-48 22.4-48 48v80h512v-80c0-25.6-22.4-48-48-48zm-144 0H192V96h128v32z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <p>
              Student
            </p>
          </span>
        </div>
      </div>
    </div>
    <div
      className="shortlistDeleteButtonContainer"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <p>
          Remove
        </p>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-trash-alt fa-w-14 "
          data-icon="trash-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
  </div>
  <div
    className="propertyList"
  >
    <div
      className="propertyContainer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="-1"
    >
      <div
        className="propertyImage"
      >
        <h4>
          IMAGE
        </h4>
      </div>
      <div
        className="propertyInfoContainer"
      >
        <div
          className="propertyTitle"
        >
          <h3>
            Non in beatae unde nostrum hic.
          </h3>
          <span
            className="propertyPrice"
          >
            £
            1475
          </span>
        </div>
        <div
          className="propertyLocation"
        >
          <span
            className="propertyLocationIcon"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-map-marker-alt fa-w-12 "
              data-icon="map-marker-alt"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M172.268 501.67C26.97 291.031 0 269.413 0 192 0 85.961 85.961 0 192 0s192 85.961 192 192c0 77.413-26.97 99.031-172.268 309.67-9.535 13.774-29.93 13.773-39.464 0zM192 272c44.183 0 80-35.817 80-80s-35.817-80-80-80-80 35.817-80 80 35.817 80 80 80z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </span>
          <span
            className="propertyAddress"
          >
            <p>
              999 Graham Crossing
              ,
               
              Bayswater
            </p>
          </span>
        </div>
        <div
          className="propertyDetails"
        >
          <span
            className="propertyFreeRooms"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-bed fa-w-20 "
              data-icon="bed"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M176 256c44.11 0 80-35.89 80-80s-35.89-80-80-80-80 35.89-80 80 35.89 80 80 80zm352-128H304c-8.84 0-16 7.16-16 16v144H64V80c0-8.84-7.16-16-16-16H16C7.16 64 0 71.16 0 80v352c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16v-48h512v48c0 8.84 7.16 16 16 16h32c8.84 0 16-7.16 16-16V240c0-61.86-50.14-112-112-112z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <p>
              5
               
              Rooms
            </p>
          </span>
          <span
            className="propertyFlatmates"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-users fa-w-20 "
              data-icon="users"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 640 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <p>
              4
               
              Flatmates
            </p>
          </span>
          <span>
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-briefcase fa-w-16 "
              data-icon="briefcase"
              data-prefix="fas"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M320 336c0 8.84-7.16 16-16 16h-96c-8.84 0-16-7.16-16-16v-48H0v144c0 25.6 22.4 48 48 48h416c25.6 0 48-22.4 48-48V288H320v48zm144-208h-80V80c0-25.6-22.4-48-48-48H176c-25.6 0-48 22.4-48 48v48H48c-25.6 0-48 22.4-48 48v80h512v-80c0-25.6-22.4-48-48-48zm-144 0H192V96h128v32z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
            <p>
              Professional
            </p>
          </span>
        </div>
      </div>
    </div>
    <div
      className="shortlistDeleteButtonContainer"
    >
      <button
        onClick={[Function]}
        type="button"
      >
        <p>
          Remove
        </p>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-trash-alt fa-w-14 "
          data-icon="trash-alt"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
